state/MapSearchProvider/MapSearchProvider.tsx

```diff
@@ -0,0 +1,37 @@
+import { Position } from 'geojson'
+import { createContext, PropsWithChildren, useCallback, useState } from 'react'
+
+type MapSearchContextProps = {
+  flyToCenter: ((center: Position) => void) | null
+}
+
+export const MapSearchContext = createContext<MapSearchContextProps | null>(null)
+MapSearchContext.displayName = 'MapSearchContext'
+
+export const MapSearchUpdateContext = createContext<
+  ((newValues: Partial<MapSearchContextProps>) => void) | null
+>(null)
+MapSearchUpdateContext.displayName = 'MapSearchUpdateContext'
+
+const defaultInitialMapSearchValues: MapSearchContextProps = {
+  flyToCenter: null,
+}
+
+const MapSearchProvider = ({ children }: PropsWithChildren) => {
+  const [values, setValues] = useState<MapSearchContextProps>(defaultInitialMapSearchValues)
+
+  const onMapSearchProviderUpdate = useCallback(
```

Comment:
if there won't be any other things in MapSearchContext there is no need to use this kind of state setting :) just put setValues as a value of context and it shouldn't rerender on change